src/client/Client.cc

```diff
@@ -1198,8 +1206,9 @@ void Client::insert_readdir_results(MetaRequest *request, MetaSession *session,
 
       ldout(cct, 15) << "" << i << ": '" << dname << "'" << dendl;
 
+      bufferlist alternate_name;
```

Comment:
I'm not a fan of putting the `alternate_name` in the xattr map like this. Is it crazy to put it in the `LeaseStat` structure which is decoded above? Was this not discussed?

Reply:
As @ukernel also suggest this, I will switch to use the LeaseStat, which will make the code simpler too.